hedera-node/src/main/java/com/hedera/services/queries/consensus/GetTopicInfoAnswer.java

```diff
@@ -107,40 +102,19 @@ public Response responseGiven(Query query, StateView view, ResponseCodeEnum vali
 			if (type == COST_ANSWER) {
 				response.setHeader(costAnswerHeader(OK, cost));
 			} else {
-				ConsensusTopicInfo.Builder info = infoBuilder(op, view);
-				response.setHeader(answerOnlyHeader(OK));
-				response.setTopicInfo(info);
+				var optionalInfo = view.infoForTopic(op.getTopicID());
+				if (optionalInfo.isPresent()){
+					response.setHeader(answerOnlyHeader(OK));
+					response.setTopicInfo(optionalInfo.get());
+				} else {
+					response.setHeader(answerOnlyHeader(FAIL_INVALID));
```

Comment:
I think `INVALID_TOPIC_ID` would make the most sense here; `FAIL_INVALID` should be reserved for an internal failure like an unhandled exception.

Reply:
Agreed. Modified to `INVALID_TOPIC_ID`